plone/app/content/browser/vocabulary.py

```diff
@@ -183,6 +184,8 @@ def __call__(self):
         ]
         if attributes:
             base_path = getNavigationRoot(context)
+            sm = getSecurityManager()
+            can_edit = sm.checkPermission('Modify portal content', context)
             for vocab_item in results:
```

Comment:
Having this info exposed could be considered unsafe in some use cases. What I did, was instead of never exposing those unsafe metadata, they are now exposed if the user has "Modify portal content" - on the context itself. Before, these metadata columns were just skipped.

I'm fine with these columns being unsafe, but maybe @vangheem can give some more insight. He added it in:

https://github.com/plone/plone.app.widgets/commit/5055725a2295422b1718ddbfe66d40870280d53a#diff-83a9a1cb178e6b38717c2427eac08433R27
and
https://github.com/plone/plone.app.widgets/commit/fd5c8cfdb12839ca16eaf4144bf8d41f4108c5b1#diff-cc8621ac4fb1caf4495921abbc02f895R33